crates/aptos-telemetry-service/src/clients/humio.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::anyhow;
+use flate2::{write::GzEncoder, Compression};
+use reqwest::{self, Url};
+
+use crate::types::humio::UnstructuredLog;
+
+pub const PEER_ID_FIELD_NAME: &str = "peer_id";
+pub const EPOCH_FIELD_NAME: &str = "epoch";
+pub const PEER_ROLE_TAG_NAME: &str = "peer_role";
+pub const CHAIN_ID_TAG_NAME: &str = "chain_id";
+
+#[derive(Clone)]
+pub struct IngestClient {
+    inner: reqwest::Client,
+    base_url: Url,
+    auth_token: String,
+}
+
+impl IngestClient {
+    pub fn new(base_url: Url, auth_token: String) -> Self {
+        Self {
+            inner: reqwest::Client::new(),
+            base_url,
+            auth_token,
+        }
+    }
+
+    pub async fn ingest_unstructured_logs(
+        &self,
+        unstructured_logs: &Vec<UnstructuredLog>,
```

Comment:
UnstructuredLog already has a vector of messages. Is it necessary for this to be a vector? Will this API ever take more than one UnstructuredLog?

Reply:
Yes, the API only takes in a vector of UnstructuredLog and I learnt it the very hard way lol. You could possibly send it a vector of items as well and it will happily ingest them.

Reply:
Should this function take just the unstructured log and then internally wrap with a vector? Just a nit really, but we can hide the nastiness as close to the source as possible.

Reply:
sounds good.